src/Teams/beta/custom/SetMgBetaChatRscConfiguration_Update.cs

```diff
@@ -0,0 +1,380 @@
+
+                        if (((Microsoft.Graph.Beta.PowerShell.Runtime.IEventListener)this).Token.IsCancellationRequested) { return; }
+
+                        // Enable preapproval configs.
+                        if (!authorizationPolicy.DefaultUserRolePermissions.PermissionGrantPoliciesAssigned.Contains(RscConfigurationConverter.MicrosoftCreatedPermissionGrantPolicyForChatRscPreApproval, StringComparer.OrdinalIgnoreCase))
+                        {
+                            IEnumerable<string> updatedPermissionGrantPolicies = authorizationPolicy.DefaultUserRolePermissions.PermissionGrantPoliciesAssigned.Union(
+                                new string[] { RscConfigurationConverter.MicrosoftCreatedPermissionGrantPolicyForChatRscPreApproval },
+                                StringComparer.OrdinalIgnoreCase);
+                            await this.Client.UpdateDefaultUserRolePermissionGrantPoliciesAssigned(
+                                updatedPermissionGrantPolicies,
+                                this,
+                                Pipeline);
+
+                            WriteVerbose($"Updated permission grant policies assigned to default user role: '{string.Join(", ", updatedPermissionGrantPolicies)}'.");
+                        }
+
+                        if (((Microsoft.Graph.Beta.PowerShell.Runtime.IEventListener)this).Token.IsCancellationRequested) { return; }
+                    }
+                    else if (this.State == MicrosoftGraphRscConfigurationState.EnabledForAllApps)
+                    {
+                        // Disable preapproval configs.
+                        if (authorizationPolicy.DefaultUserRolePermissions.PermissionGrantPoliciesAssigned.Contains(RscConfigurationConverter.MicrosoftCreatedPermissionGrantPolicyForChatRscPreApproval, StringComparer.OrdinalIgnoreCase))
+                        {
+                            IEnumerable<string> updatedPermissionGrantPolicies = authorizationPolicy.DefaultUserRolePermissions.PermissionGrantPoliciesAssigned.Except(
+                                new string[] { RscConfigurationConverter.MicrosoftCreatedPermissionGrantPolicyForChatRscPreApproval },
+                                StringComparer.OrdinalIgnoreCase);
+                            await this.Client.UpdateDefaultUserRolePermissionGrantPoliciesAssigned(
+                                updatedPermissionGrantPolicies,
+                                this,
+                                Pipeline);
+
+                            WriteVerbose($"Updated permission grant policies assigned to default user role: '{string.Join(", ", updatedPermissionGrantPolicies)}'.");
+                        }
+
+                        if (((Microsoft.Graph.Beta.PowerShell.Runtime.IEventListener)this).Token.IsCancellationRequested) { return; }
+
+                        // Enable chat RSC Teams Setting.
+                        await this.Client.UpdateTeamsAppSettings(
```

Comment:
Let's flip the order here. 

1. Enable global switch first.
2. Remove preapproval policy last.